src/Sarif/Visitors/PerRunPerRuleSplittingVisitor.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Sarif.Visitors
+{
+    public class PerRunPerRuleSplittingVisitor : SarifRewritingVisitor
```

Comment:
>PerRunPerRuleSplittingVisitor [](start = 17, length = 29)

Note that our actual intended splitting strategy is per run per target per rule. That will come in the next change.

Reply:
thanks for the clarification because it confused me. so current implementation will log a single bug for every unique issue we are seeing across all 150+ build definitions (for testing).

PS: i wonder if the word "perRun" is adding any value here? wouldn't the bugs be logged always on the per run basis? (assuming the filtering strategy filters out the new ones)

---
In reply to: [308470724](https://github.com/microsoft/sarif-sdk/pull/1596#discussion_r308470724) [](ancestors = 308470724)